spec/ParseQuery.spec.js

```diff
@@ -2816,4 +2821,363 @@ describe('Parse.Query testing', () => {
+        json: true,
+        headers: {
+          'X-Parse-Application-Id': 'test',
+          'X-Parse-REST-API-Key': 'test'
+        }
+      });
+    }).then(() => {
+      const where = {
+        subject: {
+          $text: {
+            $search: {
+              $term: 'coffee'
+            }
+          }
+        }
+      };
+      return rp.post({
+        url: 'http://localhost:8378/1/classes/TestObject',
+        json: { where, '_method': 'GET' },
+        headers: {
+          'X-Parse-Application-Id': 'test',
+          'X-Parse-REST-API-Key': 'test'
+        }
+      });
+    }).then((resp) => {
+      fail(`Text Index should not exist: ${JSON.stringify(resp)}`);
+      done();
+    }).catch((err) => {
+      expect(err.error.code).toEqual(Parse.Error.INTERNAL_SERVER_ERROR);
+      done();
+    });
+  });
+
+  it('fullTextSearch: $search', (done) => {
+    fullTextHelper().then(() => {
+      const where = {
+        subject: {
+          $text: {
+            $search: {
+              $term: 'coffee'
+            }
+          }
+        }
+      };
+      return rp.post({
+        url: 'http://localhost:8378/1/classes/TestObject',
+        json: { where, '_method': 'GET' },
+        headers: {
+          'X-Parse-Application-Id': 'test',
+          'X-Parse-REST-API-Key': 'test'
+        }
+      });
+    }).then((resp) => {
+      expect(resp.results.length).toBe(3);
+      done();
+    }, done.fail);
+  });
+
+  it('fullTextSearch: $search, sort', (done) => {
+    fullTextHelper().then(() => {
+      const where = {
+        subject: {
+          $text: {
+            $search: {
+              $term: 'coffee'
+            }
+          }
+        }
+      };
+      const order = '$score';
+      const keys = '$score';
+      return rp.post({
+        url: 'http://localhost:8378/1/classes/TestObject',
+        json: { where, order, keys, '_method': 'GET' },
+        headers: {
+          'X-Parse-Application-Id': 'test',
+          'X-Parse-REST-API-Key': 'test'
+        }
+      });
+    }).then((resp) => {
+      expect(resp.results.length).toBe(3);
+      expect(resp.results[0].score);
+      expect(resp.results[1].score);
+      expect(resp.results[2].score);
+      done();
+    }, done.fail);
+  });
+
+  it('fullTextSearch: $language', (done) => {
+    fullTextHelper().then(() => {
+      const where = {
+        subject: {
+          $text: {
+            $search: {
+              $term: 'leche',
+              $language: 'spanish'
+            }
+          }
+        }
+      };
+      return rp.post({
+        url: 'http://localhost:8378/1/classes/TestObject',
+        json: { where, '_method': 'GET' },
+        headers: {
+          'X-Parse-Application-Id': 'test',
+          'X-Parse-REST-API-Key': 'test'
+        }
+      });
+    }).then((resp) => {
+      expect(resp.results.length).toBe(2);
+      done();
+    }, done.fail);
+  });
+
+  it_exclude_dbs(['postgres'])('fullTextSearch: $caseSensitive', (done) => {
```

Comment:
There are 3 excludes here. I've added comments in the postgres adapter for $caseSensitive and $diraticSensitive.

No Index... You don't need index to do PG
Case Sensitive... Full text doesn't support case sensitive on PG. You could use regex or make a second column for lower case string.
Diratic Sensitive... Can be automated on a per table / class basis based on setup and index on PG.

Should I replace the comments with error instructions in `postgresAdapter` or leave it be?